File: Signiture-Backend/kotlin/delivery/api/src/main/java/pcy/study/api/common/api/error/UserOrderErrorCode.java
package pcy.study.api.common.api.error;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import pcy.study.api.common.api.ApiCode;

@Getter
@RequiredArgsConstructor
public enum UserOrderErrorCode implements ApiCode {

    USER_ORDER_NOT_FOUND(HttpStatus.BAD_REQUEST, 5001, "주문 정보를 찾을 수 없습니다."),
    ;

    private final HttpStatus httpStatus;
    private final Integer code;
    private final String message;
}

File: Signiture-Backend/kotlin/delivery/common/src/main/kotlin/pcy/study/common/api/code/UserOrderErrorCode.kt
package pcy.study.common.api.code

import pcy.study.common.api.ApiCode

enum class UserOrderErrorCode(
    private val httpStatusCode: Int,
    private val code: Int,
    private val message: String
) : ApiCode {

    USER_ORDER_NOT_FOUND(400, 5001, "주문 정보를 찾을 수 없습니다."),
    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getCode(): Int {
        return this.code
    }

    override fun getMessage(): String {
        return message
    }
}